play-ws-standalone-json/src/test/scala/play/api/libs/ws/JsonBodyReadablesSpec.scala

```diff
@@ -86,6 +91,15 @@ class JsonBodyReadablesSpec extends Specification with MustMatchers {
       val value: JsValue = readables.readableAsJson.transform(new StubResponse(json.getBytes("UTF-8")))
       value.toString() must beEqualTo("[]")
     }
+
+    "read an encoding of ISO-8859-1" in {
+      val readables = new JsonBodyReadables {}
+      val json = io.Source.fromResource("test.json")(Codec.ISO8859).getLines.mkString
+      val mockResponse = mock[StandaloneWSResponse]
+      Mockito.when(mockResponse.body).thenReturn(json)
+      val value: JsValue = readables.readableAsJson.transform(mockResponse)
+      value.toString must beEqualTo(json)
+    }
```

Comment:
This test is not testing what we think it is. Sure, it is testing that the new code is not blowing up. However, it's a little bit misleading. 


The Scala API works a little different then the Java one. It's also related to the place this test is being added. We are mocking `StandaloneWSResponse` and configuring the mock to return the right payload, but we are not exercising the whole test behind. 

The Java variant, creates a mock for `Response` and then uses `StandaloneAhcWSResponse`. As a result we touch some code paths that we are not exercising here. 

For instance, if we add the folloing here, the test won't pass: 

```scala
 Mockito.when(mockResponse.contentType).thenReturn("application/json;charset=iso-8859-1")
```

That's because we are not really touching the machinery that will guarantee the charset is used in the parsing. The production code will on the other hand.